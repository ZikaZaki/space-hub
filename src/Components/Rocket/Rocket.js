import React from 'react';
import PropTypes from 'prop-types';
import { useDispatch } from 'react-redux';
import { toggleRocket } from '../../redux/rockets/rockets';
// import styles
import styles from './Rocket.module.css';

const Rocket = ({
  id,
  name,
  description,
  image,
  reserved,
}) => {
  const dispatch = useDispatch();

  return (
    <div className={styles['rocket-container']}>
      <img className={styles['rocket-img']} src={image} alt="rocket" />
      <div className={styles['rocket-info']}>
        <div className={styles['rocket-name']}>
          <h2>{name}</h2>
          {reserved && <span className={styles.badge}>Reserved</span>}
        </div>
        <p>{description}</p>
        {reserved ? <button type="button" className={styles['cancel-btn']} onClick={() => dispatch(toggleRocket(id))}>Cancel Reservation</button>
          : <button type="button" className={styles['reserve-btn']} onClick={() => dispatch(toggleRocket(id))}>Reserve Rocket</button>}
      </div>
    </div>
  );
};

Rocket.propTypes = {
  id: PropTypes.number.isRequired,
  name: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
  reserved: PropTypes.bool.isRequired,
};

export default Rocket;
