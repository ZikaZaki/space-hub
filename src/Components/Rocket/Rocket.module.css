.rocket-container {
  display: flex;
  flex-direction: row;
  justify-content: flex-start;
  align-items: flex-start;
  gap: 1.5rem;
  width: 100%;
  padding: 1rem;
  margin: 0;
}

.rocket-img {
  min-width: 16rem;
  min-height: 12rem;
  object-fit: cover;
}

.rocket-info {
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  align-items: flex-start;
  gap: 0.5rem;
  width: 100%;
  height: 100%;
  padding: 0;
  margin: 0;
}

.rocket-info h2 {
  text-align: justify;
  padding: 0.2rem;
  margin: 0;
}

.rocket-info p {
  text-align: justify;
  padding: 0.2rem;
  margin: 0;
}

.badge {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  text-align: center;
  color: #fff;
  background-color: #1b9fa9;
  border-radius: 0.5rem;
  width: 5rem;
  height: 1.3rem;
  padding: 0;
  margin: 0;
}

.reserve-btn {
  text-align: center;
  color: #fff;
  background-color: #097af4;
  border-radius: 8px;
  border: none;
  outline: none;
  min-width: 10rem;
  min-height: 2.5rem;
  padding: 0;
  margin: 0;
}
