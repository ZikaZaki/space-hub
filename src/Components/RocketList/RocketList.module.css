.rocket-list-container {
  display: block;
  gap: 1.5rem;
  text-align: center;
  border-radius: 4px;
  box-shadow: 0 0 60px 0 rgba(0, 0, 0, 0.05);
  background-color: #fafafa;
  min-height: 100vh;
  height: 100%;
  width: 100%;
  padding: 0 2rem;
  margin: 0;
}

.rocket-list {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  text-align: center;
  list-style: none;
  width: 100%;
  padding: 0;
  margin: 0;
}

.rocket-item {
  width: 100%;
  margin: 0;
  padding: 0;
}
