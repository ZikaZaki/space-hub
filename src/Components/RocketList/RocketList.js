import React from 'react';
import { useSelector } from 'react-redux';
import Rocket from '../Rocket/Rocket';
// import styles
import styles from './RocketList.module.css';

const RocketList = () => {
  // Get rockets from the store
  const rockets = useSelector((state) => state.rockets);

  return (
    <div className={styles['rocket-list-container']}>
      <ul className={styles['rocket-list']}>
        {rockets && rockets.map((rocket) => (
          <li key={rocket.id} className={styles['rocket-item']}>
            <Rocket
              id={rocket.id}
              name={rocket.name}
              description={rocket.description}
              image={rocket.image}
              reserved={rocket.reserved}
            />
          </li>
        ))}
      </ul>
    </div>
  );
};

export default RocketList;
