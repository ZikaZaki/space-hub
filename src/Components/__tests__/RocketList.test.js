import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import Rocket from '../Rocket/Rocket';
import RocketList from '../RocketList/RocketList';
import store from '../../redux/configureStore';
import '@testing-library/jest-dom';

it('Check if the component renders correctly', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <RocketList>
        <Rocket
          id={1}
          name="Falcon 1"
          description="The Falcon 1 was an expendable launch system privately developed and manufactured by SpaceX during 2006-2009"
          image="https://images2.imgbox.com/3c/0e/T8iJcSN3_o.png"
          reserved={false}
        />
      </RocketList>
    </Provider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('the component should render a div container', () => {
  render(
    <Provider store={store}>
      <RocketList>
        <Rocket
          id={1}
          name="Falcon 1"
          description="The Falcon 1 was an expendable launch system privately developed and manufactured by SpaceX during 2006-2009"
          image="https://images2.imgbox.com/3c/0e/T8iJcSN3_o.png"
          reserved={false}
        />
      </RocketList>
    </Provider>,
  );

  const rocket = screen.getByTestId('rocket-list-container');

  expect(rocket).toBeInTheDocument();
});
