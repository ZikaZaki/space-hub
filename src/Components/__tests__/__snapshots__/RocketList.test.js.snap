// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check if the component renders correctly 1`] = `
<div
  className="rocket-list-container"
  data-testid="rocket-list-container"
>
  <ul
    className="rocket-list"
  />
</div>
`;
